src/App.jsx

```diff
@@ -0,0 +1,29 @@
+import { createContext } from "react";
+import "./App.css";
+import Header from "./Components/Header/Header";
+import NavigationMenu from "./Components/NavigationMenu/NavigationMenu";
+import PostFeed from "./Components/PostFeed/PostFeed";
+
+import { Navigate, Route, Routes } from "react-router-dom";
+import PostPage from "./Components/PostPage/PostPage";
+
+export const PostsContext = createContext();
```

Comment:
you are creating the context here but aren't using it here as Context.Provider ...